trunk/SUAVE/Sizing/Sizing_Loop.py

```diff
@@ -37,29 +40,50 @@ def __defaults__(self):
         #parameters common to all methods
         self.tolerance             = None
         self.initial_step          = None  #'Default', 'Table', 'SVR', 'GradientBoosting', ExtraTrees', 'RandomForest', 'Bagging', 'GPR', 'RANSAC', 'Neighbors'  
-        self.update_method         = None  #'successive_substitution', 'newton-raphson', broyden, 'damped_newton'
+        self.update_method         = None  #'successive_substitution', 'newton-raphson', 'broyden'
         self.default_y             = None  #default inputs in case the guess is very far from 
         self.default_scaling       = None  #scaling value to make sizing parameters ~1
         self.maximum_iterations    = None  #cutoff point for sizing loop to close
         self.output_filename       = None  #stores optimization parameters and closed sizing parameters
         self.sizing_evaluation     = None  #defined in the Procedure script
-        self.write_threshhold      = 9     #number of iterations before it writes, regardless of how close it is to currently written values (i.e. this step is hard to converge)
+        self.write_threshhold      = 3     #number of iterations before it writes, regardless of how close it is to currently written values (i.e. this step is hard to converge)
+        self.max_y                 = None  #vector of highest allowable y values 
+        self.min_y                 = None  #vector of lowest allowable y values
+        self.hard_max_bound        = False #set to true if you want the solver to backtrack if it reaches this bound, otherwise, just don't allow to start higher than this value
+        self.hard_min_bound        = True  #set to true if you want the solver to backtrack if it reaches this bound, otherwise, just don't allow to start lower than this value
+        self.write_threshhold      = 3     #number of iterations before it writes,
+        self.write_residuals       = False  #set to True to write the residuals at every iteration
+        self.residual_filename     = 'y_err_values.txt'
         
         #parameters that may only apply to certain methods
         self.iteration_options     = Data()
         self.iteration_options.newton_raphson_tolerance          = 5E-2             #threshhold of convergence when you start using newton raphson
         self.iteration_options.max_newton_raphson_tolerance      = 2E-3             #threshhold at which newton raphson is no longer used (to prevent overshoot and extra iterations)
         self.iteration_options.h                                 = 1E-6             #finite difference step for Newton iteration
         self.iteration_options.initialize_jacobian               = 'newton-raphson' #how Jacobian is initialized for broyden; newton-raphson by default
+        self.iteration_options.jacobian                          = np.array([np.nan])
```

Comment:
why is this defaulted to nan?

Reply:
I wanted to make sure that the code breaks if the Jacobian wasn't properly initialized